src/consensus/dkg.rs

```diff
@@ -52,8 +53,18 @@ impl DkgKey {
 
         hasher.update(&elders_info.prefix.name().0);
         hasher.update(&elders_info.prefix.bit_count().to_le_bytes());
+        hasher.finalize(&mut output);
+
+        Self(output)
+    }
 
+    pub fn retry(&self) -> Self {
```

Comment:
I am not quite sure about this part.
The DkgKey of a voting session is supposed to remain same across the nodes.
However, this `retry` stuff actually makes different version of DkgKey, depends on how many times of failure that a paticipant experienced during initialize stage.
I don't get the point why it is needed, and how it works.

Reply:
If a DKG fails, then the DKG participant doesn't necessarily have to notify the elders about it, because all they would do is to vote to restart it. So we can simplify this and the DKG participant can already restart it himself. However, to prevent the messages of the new restarted DKG to mix with the old DKG, we use a unique DKG key. This new key is deterministically derived from the old key, so all the participants will eventually use the same key. 

The participants will keep doing this until either the DKG succeeds or until a new DKG is started by the current elders (when the elder candidates change).

I should probably add a comment to this function explaining this.

Reply:
>I should probably add a comment to this function explaining this.

yes. will be good to do so